enterprise/models.py

```diff
@@ -1055,38 +1082,65 @@ def get_paginated_content(self, query_parameters):
+            content_id_values (list): If provided, return only the content IDs that are members of this list
+                                      and members of the catalog.
 
         Returns:
-            bool: True if this catalog contains the given content items, else false.
+            list: The list of content IDs that are members of the catalog.
         """
-        # Check the content_filter defined for this catalog to see if the
-        # unique key is a part of the filter that defines this catalog.
-        content_ids_in_catalog = set(self.content_filter.get(content_id_field_name, []))
+        updated_content_filter = self.content_filter.copy()
+        updated_content_filter[content_id_field_name] = content_id_values
+        results = CourseCatalogApiServiceClient().get_search_results(updated_content_filter) or []
+        return {x[content_id_field_name] for x in results}
+
+    def contains_courses(self, content_ids):
+        """
+        Return true if this catalog contains the given courses.
+
+        The content_ids parameter should be a list containing course keys
+        and/or course run ids.
+        """
+        # Translate any provided course run IDs to course keys.
+        course_keys = {parse_course_key(k) for k in content_ids}
+
+        content_ids_in_catalog = self.content_filter_ids
         if not content_ids_in_catalog:
-            # Otherwise add the unique key values to the content_filter and query
-            # the discovery service for the existence of the content.
-            updated_content_filter = self.content_filter.copy()
-            updated_content_filter[content_id_field_name] = content_id_values
-            results = CourseCatalogApiServiceClient().get_search_results(updated_content_filter)
-            if results:
-                for content in results:
-                    if content['content_type'] == 'courserun' and content['has_enrollable_seats']:
-                        content_ids_in_catalog.add(content[content_id_field_name])
-                    elif content['content_type'] == 'program' and content['is_program_eligible_for_one_click_purchase']:
-                        content_ids_in_catalog.add(content[content_id_field_name])
-
-        # Diff the content IDs found in the catalog with the set of
-        # IDs which we are checking for existence.
-        intersection = set(content_id_values) & content_ids_in_catalog
-        return len(intersection) == len(content_id_values)
+            content_ids_in_catalog = self._filter_members('key', course_keys)
+
+        return set(content_ids).issubset(content_ids_in_catalog) or course_keys.issubset(content_ids_in_catalog)
+
+    def contains_programs(self, program_uuids):
+        """
+        Return true if this catalog contains the given programs.
+        """
+        content_ids_in_catalog = self.content_filter_ids
+        if not content_ids_in_catalog:
```

Comment:
does this mean that an empty catalog is the same as a catalog with all content when determining if some program/course is part of the catalog?

Reply:
No, this is checking to see if the content_filter was defined with any content IDs (i.e. program uuids or course/courserun keys). If it was, then we can just check that list of ids to see if the provided program UUIDs are included in that list. If not, then we we add the give program UUIDs to the content_filter and make the search request to the discovery service.